### PR TITLE
#15: Implement 'runs.get_segmentation_measurements` db function

### PR DESCRIPTION
**Pull request by TerezaHrubesova**

### CODE REVIEW

File: database/src/main/runs/get_segmentation_measurements.sql [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    OUT checkpoint_name                              TEXT,
+    OUT workflow_name                                TEXT,
+    OUT checkpoint_process_start_time                TIMESTAMP WITH TIME ZONE,
+    OUT checkpoint_process_end_time                  TIMESTAMP WITH TIME ZONE,
+    OUT checkpoint_created_by                        TEXT,
+    OUT checkpoint_created_at                        TIMESTAMP WITH TIME ZONE,
+    OUT measure_type                                 TEXT,
+    OUT measure_fields                               TEXT[],
+    OUT checkpoint_measure_definition_created_by     TEXT,
+    OUT checkpoint_measure_definition_created_at     TIMESTAMP WITH TIME ZONE
```
Comment (benedeki) [inline database/src/main/runs/get_segmentation_measurements.sql:32]: Two remarks: what is returned and which order.
On the first one, I would return only values expected to be useful to the caller based on the projected use of the function. Particularly in the first iteration. It's easy to add fields later if needed.
Second, I would order the fields in some logical manner, from top to bottom, summary to detail.
So I would keep the fields:
`id_measurement, checkpoint_name, workflow_name, checkpoint_process_start_time, checkpoint_process_end_time,  measure_type, measure_fields, value`

File: database/src/main/runs/get_segmentation_measurements.sql [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      ON ch.id_checkpoint = m.key_checkpoint
+    INNER JOIN runs.checkpoints_measure_definitions chmd
+      ON m.key_checkpoint_measure_definition = chmd.id_checkpoint_measure_definition
+    WHERE ch.key_segmentation = _key_segmentation;
+
+END;
+$$
+LANGUAGE plpgsql VOLATILE SECURITY DEFINER;
+
+GRANT EXECUTE ON FUNCTION runs.get_segmentation_measurements(HSTORE) TO atum_user;
```
Comment (benedeki) [inline database/src/main/runs/get_segmentation_measurements.sql:108]: EOL

File: database/src/main/runs/get_segmentation_measurements.sql [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      chmd.measure_fields,
+      chmd.created_by AS checkpoint_measure_definition_created_by,
+      chmd.created_at AS checkpoint_measure_definition_created_at
+    FROM runs.checkpoints ch
+    INNER JOIN runs.measurements m
+      ON ch.id_checkpoint = m.key_checkpoint
+    INNER JOIN runs.checkpoints_measure_definitions chmd
+      ON m.key_checkpoint_measure_definition = chmd.id_checkpoint_measure_definition
+    WHERE ch.key_segmentation = _key_segmentation;
+
```
Comment (benedeki) [inline database/src/main/runs/get_segmentation_measurements.sql:103]: **Code style:** It's nice to end the _plpgsql_ functions with a `RETURN;` to signal all function is done (yeah it has no practical effect here).

File: database/src/main/runs/get_segmentation_measurements.sql [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+      ch.workflow_name,
+      ch.process_start_time AS checkpoint_process_start_time,
+      ch.process_end_time AS checkpoint_process_end_time,
+      ch.created_by AS checkpoint_created_by,
+      ch.created_at AS checkpoint_created_at,
+      chmd.measure_type,
+      chmd.measure_fields,
+      chmd.created_by AS checkpoint_measure_definition_created_by,
+      chmd.created_at AS checkpoint_measure_definition_created_at
+    FROM runs.checkpoints ch
```
Comment (benedeki) [inline database/src/main/runs/get_segmentation_measurements.sql:97]: Just a remark - funny, the cultural influence, that _ch_ is perceived as one letter for Czechs 😄

**Review by benedeki — CHANGES_REQUESTED** [review]
It's great work.
Sorry for requesting changes on such rather small things. But as it's to be introduction to the DB code style... 👼

File: database/src/main/runs/get_segmentation_measurements.sql (hunk 1 above)
Comment (TerezaHrubesova) [reply database/src/main/runs/get_segmentation_measurements.sql]: Updated.

File: database/src/main/runs/get_segmentation_measurements.sql (hunk 2 above)
Comment (TerezaHrubesova) [reply database/src/main/runs/get_segmentation_measurements.sql]: Added.

File: database/src/main/runs/get_segmentation_measurements.sql (hunk 3 above)
Comment (TerezaHrubesova) [reply database/src/main/runs/get_segmentation_measurements.sql:85]: Added.

**Review by benedeki — APPROVED** [review]
Thank you looks great!

Please don't merge yet. Let's merge it into master when `feature/first-dataft-od-db-data-model-v2` will have been merged.

File: database/src/main/runs/get_segmentation_measurements.sql (hunk 4 above)
Comment (benedeki) [reply database/src/main/runs/get_segmentation_measurements.sql]: It was not meant as a impulse for change. It was just a funny (or geeky) observation. 😄

**Review by dk1844 — DISMISSED** [review]
LGTM (just read the code)

**Review by benedeki — APPROVED** [review]

**Pull request merged**